packages/templates/typescript-react-apollo/src/helpers/generate-fragment.ts

```diff
@@ -12,12 +12,12 @@ export function generateFragment(fragment: any, options: any): string {
 
     if (config.noNamespaces) {
       return `
-      const ${fragment.name}Document = ${gql(fragment, options)};
+      export const get${fragment.name}Document = () => ${gql(fragment, options)};
       `;
     } else {
       return `
         export namespace ${fragment.name} {
-          export const Document = ${gql(fragment, options)};
+          export const getDocument = () => ${gql(fragment, options)};
```

Comment:
Why a getter?

Reply:
Because `Document` sometimes conflicts with the DOM `Document`

Reply:
Also it sometimes throws undefined due to the order of fragment definitions ( if PostFragment has AuthorFragment and AuthorFragment is defined after PostFragment, it throws undefined for AuthorFragment )